cs/src/core/RuntimeSchema.cs

```diff
@@ -34,25 +35,37 @@ public RuntimeSchema(SchemaDef schema)
 
         public RuntimeSchema GetBaseSchema()
         {
-            Debug.Assert(IsStruct);
+            if (!IsStruct)
+            {
+                throw new InvalidOperationException();
```

Comment:
>InvalidOperationException(); [](start = 26, length = 28)

I made this comment in Chad's PR, which got lost. Please add exception error message here and in other places in this file.

Reply:
Done.